Source/Tools/FEXLoader/LinuxSyscalls/SignalDelegator.cpp

```diff
@@ -1702,6 +1801,12 @@ namespace FEX::HLE {
 
     // Get the current host signal mask
     ::syscall(SYS_rt_sigprocmask, 0, nullptr, &ThreadData.CurrentSignalMask.Val, 8);
+
+    if (Thread != (FEXCore::Core::InternalThreadState*)UINTPTR_MAX) {
```

Comment:
TestHarnessRunner's Host runner passes in `UINPTR_MAX` for the Internal thread state object.
It was done this way because nullptr can't be passed in to that. I didn't write that code but rewriting the logic to allow nullptr through those (or creating a dummy InternalThreadState?) needs to be done at some point.
Rewriting that logic to make it cleaner would be a decent chunk of effort that shouldn't be spent on this PR.

Reply:
Sounds fair 👍